backend/dataall/modules/dataset_sharing/aws/s3_client.py

```diff
@@ -112,13 +119,37 @@ def generate_default_bucket_policy(s3_bucket_name: str):
         return policy
 
 
```

Comment:
Same comment as with KMS, this is not an AWS boto3 call, seems more suitable to place it in the share managers in the service layer